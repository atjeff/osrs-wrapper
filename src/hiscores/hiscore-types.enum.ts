export enum HiscoreTypes {
  'normal' = 'hiscore_oldschool',
  'ironman' = 'hiscore_oldschool_ironman',
  'ultimate' = 'hiscore_oldschool_ultimate',
  'hardcore' = 'hiscore_oldschool_hardcore_ironman',
  'deadman' = 'hiscore_oldschool_deadman',
  'league' = 'hiscore_oldschool_seasonal',
}
