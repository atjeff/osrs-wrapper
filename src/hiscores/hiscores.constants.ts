export const SKILL_NAMES = [
    'overall',
    'attack',
    'defence',
    'strength',
    'hitpoints',
    'ranged',
    'prayer',
    'magic',
    'cooking',
    'woodcutting',
    'fletching',
    'fishing',
    'firemaking',
    'crafting',
    'smithing',
    'mining',
    'herblore',
    'agility',
    'thieving',
    'slayer',
    'farming',
    'runecrafting',
    'hunter',
    'construction'
];

export const MINIGAME_NAMES = [
    'leaguePoints',
    'bountyHunter',
    'bountyHunterRogues',
    'clueScrollsAll',
    'clueScrollsBeginner',
    'clueScrollsEasy',
    'clueScrollsMedium',
    'clueScrollsHard',
    'clueScrollsElite',
    'clueScrollsMaster',
    'lms',
    'pvpArena',
    'soulWarsZeal',
    'riftsClosed',
];

export const BOSS_NAMES = [
    'abyssalSire',
    'alchemicalHydra',
    'barrowsChests',
    'bryophyta',
    'callisto',
    'cerberus',
    'chambersOfXeric',
    'chambersOfXericChallengeMode',
    'chaosElemental',
    'chaosFanatic',
    'commanderZilyana',
    'corporealBeast',
    'crazyArchaeologist',
    'dagannothPrime',
    'dagannothRex',
    'dagannothSupreme',
    'derangedArchaeologist',
    'generalGraardor',
    'giantMole',
    'grotesqueGuardians',
    'hespori',
    'kalphiteQueen',
    'kingBlackDragon',
    'kraken',
    'kreearra',
    'krilTsutsaroth',
    'mimic',
    'nex',
    'nightmare',
    'phosanisNightmare',
    'obor',
    'phantomMuspah',
    'sarachnis',
    'scorpia',
    'skotizo',
    'tempoross',
    'gauntlet',
    'corruptedGauntlet',
    'theatreOfBlood',
    'theatreOfBloodHardMode',
    'thermonuclearSmokeDevil',
    'tombsOfAmascut',
    'tombsOfAmascutExpertMode',
    'tzKalZuk',
    'tzTokJad',
    'venenatis',
    'vetion',
    'vorkath',
    'wintertodt',
    'zalcano',
    'zulrah'
];
