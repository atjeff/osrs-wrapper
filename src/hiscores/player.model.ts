export interface Player {
    skills: Skills;
    minigames: Minigames;
    bosses: Bosses;
}

export interface Skills {
    overall: Skill;
    attack: Skill;
    defence: Skill;
    strength: Skill;
    hitpoints: Skill;
    ranged: Skill;
    prayer: Skill;
    magic: Skill;
    cooking: Skill;
    woodcutting: Skill;
    fletching: Skill;
    fishing: Skill;
    firemaking: Skill;
    crafting: Skill;
    smithing: Skill;
    mining: Skill;
    herblore: Skill;
    agility: Skill;
    thieving: Skill;
    slayer: Skill;
    farming: Skill;
    runecrafting: Skill;
    hunter: Skill;
    construction: Skill;
}

export interface Skill {
    rank: string;
    level: string;
    xp: string;
}

export interface Minigames {
    leaguePoints: Minigame;
    bountyHunter: Minigame;
    bountyHunterRogues: Minigame;
    clueScrollsAll: Minigame;
    clueScrollsBeginner: Minigame;
    clueScrollsEasy: Minigame;
    clueScrollsMedium: Minigame;
    clueScrollsHard: Minigame;
    clueScrollsElite: Minigame;
    clueScrollsMaster: Minigame;
    lms: Minigame;
    pvpArena: Minigame;
    soulWarsZeal: Minigame;
    riftsClosed: Minigame;
}

export interface Minigame {
    rank: string;
    score: string;
}

export interface Bosses {
    abyssalSire: Boss;
    alchemicalHydra: Boss;
    barrowsChests: Boss;
    bryophyta: Boss;
    callisto: Boss;
    cerberus: Boss;
    chambersOfXeric: Boss;
    chambersOfXericChallengeMode: Boss;
    chaosElemental: Boss;
    chaosFanatic: Boss;
    commanderZilyana: Boss;
    corporealBeast: Boss;
    crazyArchaeologist: Boss;
    dagannothPrime: Boss;
    dagannothRex: Boss;
    dagannothSupreme: Boss;
    derangedArchaeologist: Boss;
    generalGraardor: Boss;
    giantMole: Boss;
    grotesqueGuardians: Boss;
    hespori: Boss;
    kalphiteQueen: Boss;
    kingBlackDragon: Boss;
    kraken: Boss;
    kreearra: Boss;
    krilTsutsaroth: Boss;
    mimic: Boss;
    nex: Boss;
    nightmare: Boss;
    phosanisNightmare: Boss;
    obor: Boss;
    phantomMuspah: Boss;
    sarachnis: Boss;
    scorpia: Boss;
    skotizo: Boss;
    tempoross: Boss;
    gauntlet: Boss;
    corruptedGauntlet: Boss;
    theatreOfBlood: Boss;
    theatreOfBloodHardMode: Boss;
    thermonuclearSmokeDevil: Boss;
    tombsOfAmascut: Boss;
    tombsOfAmascutExpertMode: Boss;
    tzKalZuk: Boss;
    tzTokJad: Boss;
    venenatis: Boss;
    vetion: Boss;
    vorkath: Boss;
    wintertodt: Boss;
    zalcano: Boss;
    zulrah: Boss;
}

export interface Boss {
    rank: string;
    kills: string;
}
