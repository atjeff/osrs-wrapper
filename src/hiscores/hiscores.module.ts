import { request } from 'gaxios';
import { HiscoreTypes } from './hiscore-types.enum';
import { BOSS_NAMES, MINIGAME_NAMES, SKILL_NAMES } from './hiscores.constants';
import { Bosses, HiscoreResponse, Minigames, Player, Skills } from './player.model';

/**
 * Fetches player hiscores for specified game type. Includes Skills, Minigames, and Boss kill counts.
 *
 * @export
 * @param {string} username
 * @param {HiscoreTypes} [type=HiscoreTypes.normal]
 * @returns {Promise<Player>}
 */
export async function getHiscores(username: string, type: HiscoreTypes = HiscoreTypes.normal): Promise<Player> {
    const { data } = await request<HiscoreResponse>({
        method: 'GET',
        url: `http://services.runescape.com/m=${type}/index_lite.json`,
        params: { player: encodeURIComponent(username) },
    });

    const skills = {} as Skills;
    for (const skill of data.skills) {
        const { rank, level, xp, name } = skill;
        const skillName = SKILL_NAMES[skill.name];
        skills[skillName] = { rank, level, xp };
    }

    const minigames = {} as Minigames;
    const bosses = {} as Bosses;
    for (const activity of data.activities) {
        const { rank, score, name } = activity;

        if (!!MINIGAME_NAMES[name]) {
            minigames[MINIGAME_NAMES[name]] = { rank, score };
        } else if (!!BOSS_NAMES[name]) {
            bosses[BOSS_NAMES[name]] = { rank, kills: score };
        }
    }

    return { skills, minigames, bosses };
}
