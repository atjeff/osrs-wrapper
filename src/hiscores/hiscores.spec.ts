import { request } from 'gaxios';
import { mockHiscoreResponse } from '../../test/hiscore-response.mock';
import { HiscoreTypes } from './hiscore-types.enum';
import { getHiscores } from './hiscores.module';

jest.mock('gaxios');

describe('Hiscores', () => {
    describe('getHiscores', () => {
        const playerName = 'Bald';
        const mockedSuccessfulResponse = { data: mockHiscoreResponse } as any;
        let mockedGaxiosRequest = request as jest.MockedFunction<typeof request>;

        mockedGaxiosRequest.mockResolvedValue(mockedSuccessfulResponse);

        afterEach(() => mockedGaxiosRequest.mockClear());
        afterAll(() => jest.unmock('gaxios'));

        it('should make a GET request', async () => {
            await getHiscores(playerName);

            expect(mockedGaxiosRequest.mock.calls[0][0]).toMatchSnapshot();
        });

        it('should make a request with the passed in player name', async () => {
            await getHiscores(playerName);

            expect(mockedGaxiosRequest.mock.calls[0][0]).toMatchSnapshot();
        });

        it('should default type to HiscoreTypes.normal', async () => {
            await getHiscores(playerName);

            expect(mockedGaxiosRequest.mock.calls[0][0]).toMatchSnapshot();
        });

        it('should hit the specified hiscore type endpoint', async () => {
            const hiscoreType = HiscoreTypes.league;
            await getHiscores(playerName, hiscoreType);

            expect(mockedGaxiosRequest.mock.calls[0][0]).toMatchSnapshot();
        });

        it('should properly parse rank, level, and xp for each skill', async () => {
            const { skills } = await getHiscores(playerName);

            expect(skills).toMatchSnapshot();
        });

        it('should properly parse rank and score for each minigame', async () => {
            const { minigames } = await getHiscores(playerName);

            expect(minigames).toMatchSnapshot();
        });

        it('should properly parse rank and kills for each boss', async () => {
            const { bosses } = await getHiscores(playerName);

            expect(bosses).toMatchSnapshot();
        });

        it('should return skills, minigames, and bosses from a json response', async () => {
            const output = await getHiscores(playerName);

            expect(output).toMatchSnapshot();
        });
    });
});
