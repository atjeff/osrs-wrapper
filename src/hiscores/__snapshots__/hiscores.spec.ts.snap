// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hiscores getHiscores should default type to HiscoreTypes.normal 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "player": "Bald",
  },
  "url": "http://services.runescape.com/m=hiscore_oldschool/index_lite.ws",
}
`;

exports[`Hiscores getHiscores should hit the specified hiscore type endpoint 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "player": "Bald",
  },
  "url": "http://services.runescape.com/m=hiscore_oldschool_seasonal/index_lite.ws",
}
`;

exports[`Hiscores getHiscores should make a GET request 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "player": "Bald",
  },
  "url": "http://services.runescape.com/m=hiscore_oldschool/index_lite.ws",
}
`;

exports[`Hiscores getHiscores should make a request with the passed in player name 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "player": "Bald",
  },
  "url": "http://services.runescape.com/m=hiscore_oldschool/index_lite.ws",
}
`;

exports[`Hiscores getHiscores should properly parse rank and kills for each boss 1`] = `
Object {
  "abyssalSire": Object {
    "kills": "250",
    "rank": "75147",
  },
  "alchemicalHydra": Object {
    "kills": "3334",
    "rank": "5032",
  },
  "barrowsChests": Object {
    "kills": "307",
    "rank": "137028",
  },
  "bryophyta": Object {
    "kills": "23",
    "rank": "34596",
  },
  "callisto": Object {
    "kills": "209",
    "rank": "20461",
  },
  "cerberus": Object {
    "kills": "3000",
    "rank": "6013",
  },
  "chambersOfXeric": Object {
    "kills": "5589",
    "rank": "60",
  },
  "chambersOfXericChallengeMode": Object {
    "kills": "1045",
    "rank": "261",
  },
  "chaosElemental": Object {
    "kills": "150",
    "rank": "25763",
  },
  "chaosFanatic": Object {
    "kills": "249",
    "rank": "13130",
  },
  "commanderZilyana": Object {
    "kills": "2347",
    "rank": "1513",
  },
  "corporealBeast": Object {
    "kills": "7930",
    "rank": "131",
  },
  "corruptedGauntlet": Object {
    "kills": "207",
    "rank": "42296",
  },
  "crazyArchaeologist": Object {
    "kills": "77",
    "rank": "57131",
  },
  "dagannothPrime": Object {
    "kills": "5644",
    "rank": "377",
  },
  "dagannothRex": Object {
    "kills": "6851",
    "rank": "314",
  },
  "dagannothSupreme": Object {
    "kills": "5569",
    "rank": "400",
  },
  "derangedArchaeologist": Object {
    "kills": "50",
    "rank": "17310",
  },
  "gauntlet": Object {
    "kills": "224",
    "rank": "1946",
  },
  "generalGraardor": Object {
    "kills": "10540",
    "rank": "133",
  },
  "giantMole": Object {
    "kills": "4081",
    "rank": "2613",
  },
  "grotesqueGuardians": Object {
    "kills": "5666",
    "rank": "135",
  },
  "hespori": Object {
    "kills": "231",
    "rank": "6828",
  },
  "kalphiteQueen": Object {
    "kills": "1088",
    "rank": "4952",
  },
  "kingBlackDragon": Object {
    "kills": "3938",
    "rank": "1880",
  },
  "kraken": Object {
    "kills": "10696",
    "rank": "627",
  },
  "kreearra": Object {
    "kills": "5960",
    "rank": "285",
  },
  "krilTsutsaroth": Object {
    "kills": "10793",
    "rank": "26",
  },
  "mimic": Object {
    "kills": "39",
    "rank": "277",
  },
  "nex": Object {
    "kills": "1919",
    "rank": "4908",
  },
  "nightmare": Object {
    "kills": "7231",
    "rank": "60",
  },
  "obor": Object {
    "kills": "38",
    "rank": "18065",
  },
  "phantomMuspah": Object {
    "kills": "847",
    "rank": "7",
  },
  "phosanisNightmare": Object {
    "kills": "131",
    "rank": "9589",
  },
  "sarachnis": Object {
    "kills": "1179",
    "rank": "3914",
  },
  "scorpia": Object {
    "kills": "10139",
    "rank": "8",
  },
  "skotizo": Object {
    "kills": "65",
    "rank": "14891",
  },
  "tempoross": Object {
    "kills": "226",
    "rank": "32565",
  },
  "theatreOfBlood": Object {
    "kills": "4940",
    "rank": "79",
  },
  "theatreOfBloodHardMode": Object {
    "kills": "221",
    "rank": "2301",
  },
  "thermonuclearSmokeDevil": Object {
    "kills": "2165",
    "rank": "11980",
  },
  "tombsOfAmascut": Object {
    "kills": "222",
    "rank": "3258",
  },
  "tombsOfAmascutExpertMode": Object {
    "kills": "2015",
    "rank": "1",
  },
  "tzKalZuk": Object {
    "kills": "55",
    "rank": "312",
  },
  "tzTokJad": Object {
    "kills": "94",
    "rank": "838",
  },
  "venenatis": Object {
    "kills": "1226",
    "rank": "3984",
  },
  "vetion": Object {
    "kills": "641",
    "rank": "2397",
  },
  "vorkath": Object {
    "kills": "1410",
    "rank": "31217",
  },
  "wintertodt": Object {
    "kills": "4832",
    "rank": "507",
  },
  "zalcano": Object {
    "kills": "1302",
    "rank": "2262",
  },
  "zulrah": Object {
    "kills": "11026",
    "rank": "1606",
  },
}
`;

exports[`Hiscores getHiscores should properly parse rank and score for each minigame 1`] = `
Object {
  "bountyHunter": Object {
    "rank": "-1",
    "score": "-1",
  },
  "bountyHunterRogues": Object {
    "rank": "-1",
    "score": "-1",
  },
  "clueScrollsAll": Object {
    "rank": "541",
    "score": "5637",
  },
  "clueScrollsBeginner": Object {
    "rank": "1838",
    "score": "601",
  },
  "clueScrollsEasy": Object {
    "rank": "7103",
    "score": "500",
  },
  "clueScrollsElite": Object {
    "rank": "4285",
    "score": "200",
  },
  "clueScrollsHard": Object {
    "rank": "33018",
    "score": "301",
  },
  "clueScrollsMaster": Object {
    "rank": "17",
    "score": "1534",
  },
  "clueScrollsMedium": Object {
    "rank": "690",
    "score": "2501",
  },
  "leaguePoints": Object {
    "rank": "-1",
    "score": "-1",
  },
  "lms": Object {
    "rank": "163363",
    "score": "507",
  },
  "pvpArena": Object {
    "rank": "-1",
    "score": "-1",
  },
  "riftsClosed": Object {
    "rank": "56",
    "score": "2277",
  },
  "soulWarsZeal": Object {
    "rank": "3375",
    "score": "14724",
  },
}
`;

exports[`Hiscores getHiscores should properly parse rank, level, and xp for each skill 1`] = `
Object {
  "agility": Object {
    "level": "99",
    "rank": "701",
    "xp": "36007958",
  },
  "attack": Object {
    "level": "99",
    "rank": "824",
    "xp": "104224886",
  },
  "construction": Object {
    "level": "99",
    "rank": "3117",
    "xp": "14165910",
  },
  "cooking": Object {
    "level": "99",
    "rank": "877",
    "xp": "200000000",
  },
  "crafting": Object {
    "level": "99",
    "rank": "4270",
    "xp": "17691019",
  },
  "defence": Object {
    "level": "99",
    "rank": "530",
    "xp": "108054847",
  },
  "farming": Object {
    "level": "99",
    "rank": "1101",
    "xp": "113764644",
  },
  "firemaking": Object {
    "level": "99",
    "rank": "841",
    "xp": "126998621",
  },
  "fishing": Object {
    "level": "99",
    "rank": "1704",
    "xp": "46330395",
  },
  "fletching": Object {
    "level": "99",
    "rank": "6184",
    "xp": "19525268",
  },
  "herblore": Object {
    "level": "99",
    "rank": "5090",
    "xp": "15559824",
  },
  "hitpoints": Object {
    "level": "99",
    "rank": "96",
    "xp": "200000000",
  },
  "hunter": Object {
    "level": "99",
    "rank": "1363",
    "xp": "53077068",
  },
  "magic": Object {
    "level": "99",
    "rank": "352",
    "xp": "130111372",
  },
  "mining": Object {
    "level": "99",
    "rank": "6090",
    "xp": "20743294",
  },
  "overall": Object {
    "level": "2277",
    "rank": "346",
    "xp": "1816945615",
  },
  "prayer": Object {
    "level": "99",
    "rank": "3520",
    "xp": "14709754",
  },
  "ranged": Object {
    "level": "99",
    "rank": "58",
    "xp": "200000000",
  },
  "runecrafting": Object {
    "level": "99",
    "rank": "943",
    "xp": "37806702",
  },
  "slayer": Object {
    "level": "99",
    "rank": "578",
    "xp": "63465365",
  },
  "smithing": Object {
    "level": "99",
    "rank": "15968",
    "xp": "13761910",
  },
  "strength": Object {
    "level": "99",
    "rank": "201",
    "xp": "200000000",
  },
  "thieving": Object {
    "level": "99",
    "rank": "8740",
    "xp": "22601332",
  },
  "woodcutting": Object {
    "level": "99",
    "rank": "1993",
    "xp": "58345446",
  },
}
`;

exports[`Hiscores getHiscores should return skills, minigames, and bosses from a csv 1`] = `
Object {
  "bosses": Object {
    "abyssalSire": Object {
      "kills": "250",
      "rank": "75147",
    },
    "alchemicalHydra": Object {
      "kills": "3334",
      "rank": "5032",
    },
    "barrowsChests": Object {
      "kills": "307",
      "rank": "137028",
    },
    "bryophyta": Object {
      "kills": "23",
      "rank": "34596",
    },
    "callisto": Object {
      "kills": "209",
      "rank": "20461",
    },
    "cerberus": Object {
      "kills": "3000",
      "rank": "6013",
    },
    "chambersOfXeric": Object {
      "kills": "5589",
      "rank": "60",
    },
    "chambersOfXericChallengeMode": Object {
      "kills": "1045",
      "rank": "261",
    },
    "chaosElemental": Object {
      "kills": "150",
      "rank": "25763",
    },
    "chaosFanatic": Object {
      "kills": "249",
      "rank": "13130",
    },
    "commanderZilyana": Object {
      "kills": "2347",
      "rank": "1513",
    },
    "corporealBeast": Object {
      "kills": "7930",
      "rank": "131",
    },
    "corruptedGauntlet": Object {
      "kills": "207",
      "rank": "42296",
    },
    "crazyArchaeologist": Object {
      "kills": "77",
      "rank": "57131",
    },
    "dagannothPrime": Object {
      "kills": "5644",
      "rank": "377",
    },
    "dagannothRex": Object {
      "kills": "6851",
      "rank": "314",
    },
    "dagannothSupreme": Object {
      "kills": "5569",
      "rank": "400",
    },
    "derangedArchaeologist": Object {
      "kills": "50",
      "rank": "17310",
    },
    "gauntlet": Object {
      "kills": "224",
      "rank": "1946",
    },
    "generalGraardor": Object {
      "kills": "10540",
      "rank": "133",
    },
    "giantMole": Object {
      "kills": "4081",
      "rank": "2613",
    },
    "grotesqueGuardians": Object {
      "kills": "5666",
      "rank": "135",
    },
    "hespori": Object {
      "kills": "231",
      "rank": "6828",
    },
    "kalphiteQueen": Object {
      "kills": "1088",
      "rank": "4952",
    },
    "kingBlackDragon": Object {
      "kills": "3938",
      "rank": "1880",
    },
    "kraken": Object {
      "kills": "10696",
      "rank": "627",
    },
    "kreearra": Object {
      "kills": "5960",
      "rank": "285",
    },
    "krilTsutsaroth": Object {
      "kills": "10793",
      "rank": "26",
    },
    "mimic": Object {
      "kills": "39",
      "rank": "277",
    },
    "nex": Object {
      "kills": "1919",
      "rank": "4908",
    },
    "nightmare": Object {
      "kills": "7231",
      "rank": "60",
    },
    "obor": Object {
      "kills": "38",
      "rank": "18065",
    },
    "phantomMuspah": Object {
      "kills": "847",
      "rank": "7",
    },
    "phosanisNightmare": Object {
      "kills": "131",
      "rank": "9589",
    },
    "sarachnis": Object {
      "kills": "1179",
      "rank": "3914",
    },
    "scorpia": Object {
      "kills": "10139",
      "rank": "8",
    },
    "skotizo": Object {
      "kills": "65",
      "rank": "14891",
    },
    "tempoross": Object {
      "kills": "226",
      "rank": "32565",
    },
    "theatreOfBlood": Object {
      "kills": "4940",
      "rank": "79",
    },
    "theatreOfBloodHardMode": Object {
      "kills": "221",
      "rank": "2301",
    },
    "thermonuclearSmokeDevil": Object {
      "kills": "2165",
      "rank": "11980",
    },
    "tombsOfAmascut": Object {
      "kills": "222",
      "rank": "3258",
    },
    "tombsOfAmascutExpertMode": Object {
      "kills": "2015",
      "rank": "1",
    },
    "tzKalZuk": Object {
      "kills": "55",
      "rank": "312",
    },
    "tzTokJad": Object {
      "kills": "94",
      "rank": "838",
    },
    "venenatis": Object {
      "kills": "1226",
      "rank": "3984",
    },
    "vetion": Object {
      "kills": "641",
      "rank": "2397",
    },
    "vorkath": Object {
      "kills": "1410",
      "rank": "31217",
    },
    "wintertodt": Object {
      "kills": "4832",
      "rank": "507",
    },
    "zalcano": Object {
      "kills": "1302",
      "rank": "2262",
    },
    "zulrah": Object {
      "kills": "11026",
      "rank": "1606",
    },
  },
  "minigames": Object {
    "bountyHunter": Object {
      "rank": "-1",
      "score": "-1",
    },
    "bountyHunterRogues": Object {
      "rank": "-1",
      "score": "-1",
    },
    "clueScrollsAll": Object {
      "rank": "541",
      "score": "5637",
    },
    "clueScrollsBeginner": Object {
      "rank": "1838",
      "score": "601",
    },
    "clueScrollsEasy": Object {
      "rank": "7103",
      "score": "500",
    },
    "clueScrollsElite": Object {
      "rank": "4285",
      "score": "200",
    },
    "clueScrollsHard": Object {
      "rank": "33018",
      "score": "301",
    },
    "clueScrollsMaster": Object {
      "rank": "17",
      "score": "1534",
    },
    "clueScrollsMedium": Object {
      "rank": "690",
      "score": "2501",
    },
    "leaguePoints": Object {
      "rank": "-1",
      "score": "-1",
    },
    "lms": Object {
      "rank": "163363",
      "score": "507",
    },
    "pvpArena": Object {
      "rank": "-1",
      "score": "-1",
    },
    "riftsClosed": Object {
      "rank": "56",
      "score": "2277",
    },
    "soulWarsZeal": Object {
      "rank": "3375",
      "score": "14724",
    },
  },
  "skills": Object {
    "agility": Object {
      "level": "99",
      "rank": "701",
      "xp": "36007958",
    },
    "attack": Object {
      "level": "99",
      "rank": "824",
      "xp": "104224886",
    },
    "construction": Object {
      "level": "99",
      "rank": "3117",
      "xp": "14165910",
    },
    "cooking": Object {
      "level": "99",
      "rank": "877",
      "xp": "200000000",
    },
    "crafting": Object {
      "level": "99",
      "rank": "4270",
      "xp": "17691019",
    },
    "defence": Object {
      "level": "99",
      "rank": "530",
      "xp": "108054847",
    },
    "farming": Object {
      "level": "99",
      "rank": "1101",
      "xp": "113764644",
    },
    "firemaking": Object {
      "level": "99",
      "rank": "841",
      "xp": "126998621",
    },
    "fishing": Object {
      "level": "99",
      "rank": "1704",
      "xp": "46330395",
    },
    "fletching": Object {
      "level": "99",
      "rank": "6184",
      "xp": "19525268",
    },
    "herblore": Object {
      "level": "99",
      "rank": "5090",
      "xp": "15559824",
    },
    "hitpoints": Object {
      "level": "99",
      "rank": "96",
      "xp": "200000000",
    },
    "hunter": Object {
      "level": "99",
      "rank": "1363",
      "xp": "53077068",
    },
    "magic": Object {
      "level": "99",
      "rank": "352",
      "xp": "130111372",
    },
    "mining": Object {
      "level": "99",
      "rank": "6090",
      "xp": "20743294",
    },
    "overall": Object {
      "level": "2277",
      "rank": "346",
      "xp": "1816945615",
    },
    "prayer": Object {
      "level": "99",
      "rank": "3520",
      "xp": "14709754",
    },
    "ranged": Object {
      "level": "99",
      "rank": "58",
      "xp": "200000000",
    },
    "runecrafting": Object {
      "level": "99",
      "rank": "943",
      "xp": "37806702",
    },
    "slayer": Object {
      "level": "99",
      "rank": "578",
      "xp": "63465365",
    },
    "smithing": Object {
      "level": "99",
      "rank": "15968",
      "xp": "13761910",
    },
    "strength": Object {
      "level": "99",
      "rank": "201",
      "xp": "200000000",
    },
    "thieving": Object {
      "level": "99",
      "rank": "8740",
      "xp": "22601332",
    },
    "woodcutting": Object {
      "level": "99",
      "rank": "1993",
      "xp": "58345446",
    },
  },
}
`;
